contracts/DCAHubSwapper/ThirdPartyDCAHubSwapper.sol

```diff
@@ -98,33 +98,24 @@ contract ThirdPartyDCAHubSwapper is IDCAHubSwapCallee {
   }
 
   function _approveAllowances(Allowance[] memory _allowanceTargets) internal {
-    for (uint256 i = 0; i < _allowanceTargets.length; ) {
+    for (uint256 i = 0; i < _allowanceTargets.length; ++i) {
       Allowance memory _target = _allowanceTargets[i];
       uint256 _currentAllowance = _target.token.allowance(address(this), _target.spender);
       if (_currentAllowance < _target.amount) {
```

Comment:
Let's remove the `amount` property from `Allowance` and simply execute the `forceApprove` for each allowance target. We can check if the current allowance is enough from off-chain